hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyOutputStream.java

```diff
@@ -201,7 +209,11 @@ private void handleWrite(byte[] b, int off, long len, boolean retry)
         // comes via Exception path.
         int writeLen = Math.min((int) len, (int) current.getRemaining());
         long currentPos = current.getWrittenDataLength();
-        writeToOutputStream(current, retry, len, b, writeLen, off, currentPos);
+        // writeLen will be updated based on whether the write was succeeded
+        // or if it sees an exception, how much the actual write was
+        // acknowledged.
+        writeLen = writeToOutputStream(current, retry, len, b, writeLen,
```

Comment:
writeLen, goes in both as an agrument as well as a return value to the function. Can we change this to something like expectedWritelen, and writtenLen